### PR TITLE
add docs for generated columns and using them for partitioned tables

### PR DESCRIPTION
**Pull request by mfelsche**
### do not merge while not running completely

### CODE REVIEW

File: docs/sql/ddl.txt [hunk 1]
```diff
@@ -129,6 +129,63 @@ Column names are restricted in terms of characters. They:
 
   - may not *start* with an underscore: ``_``
 
+
+.. _sql-ddl-generated-columns:
+
+Generated Columns
+=================
+
+It is possible to define additional columns whose value is computed by
```
Comment (dobe) [inline docs/sql/ddl.txt]: i would leave out additional

File: docs/sql/ddl.txt [hunk 2]
```diff
@@ -129,6 +129,63 @@ Column names are restricted in terms of characters. They:
...
+    +----------+---------+--------------+
+    SELECT 2 rows in set (... sec)
+
+If values are supplied for generated columns, these values are validated
+against the result of applying the *generation expression*::
+
+    cr> INSERT INTO computed (divident, divisor, quotient) VALUES (100.0, 2.0, 12.0);
+    ERROR NOT OK
+
+They can also be used in the :ref:`partitioned_by_clause` in order to
```
Comment (dobe) [inline docs/sql/ddl.txt]: i would either make an example with it or leave the paragraph out completelely

File: docs/sql/information_schema.txt [hunk 3]
```diff
@@ -112,20 +112,22 @@ You can even query this tables' own columns (attention: this might lead to infin
...
+    | data_type             | string    | 5                |
+    | is_generated          | boolean   | 6                |
+    | generation_expression | string    | 7                |
+    +------------------+-----------+-----------------------+
+    SELECT 7 rows in set (... sec)
 
 .. note::
 
-  Columns at Crate are always sorted alphabetically in ascending order despite in which order
+  Columns at Crate are always sorted alphabetically in ascending order regardless of the order
```
Comment (dobe) [inline docs/sql/information_schema.txt]: if we are at it. the "at Crate"

File: docs/sql/dml.txt [hunk 4]
```diff
@@ -12,7 +12,7 @@ Inserting data to ``Crate`` is done by using the SQL ``INSERT`` statement.
 .. note::
 
     The column list at ``Crate`` is always ordered alphabetically by column name
```
Comment (dobe) [inline docs/sql/dml.txt]: remove "at Crate"

**Comment by dobe** [comment]
ack

**Comment by bogensberger** [comment]
ack

**Pull request merged**